ssl/t1_trce.c

```diff
@@ -94,10 +94,10 @@ static const ssl_trace_tbl ssl_handshake_tbl[] = {
     {SSL3_MT_SERVER_DONE, "ServerHelloDone"},
     {SSL3_MT_CERTIFICATE_VERIFY, "CertificateVerify"},
     {SSL3_MT_CLIENT_KEY_EXCHANGE, "ClientKeyExchange"},
-    {SSL3_MT_CERTIFICATE_STATUS, "CertificateStatus"},
-    {SSL3_MT_CLIENT_KEY_EXCHANGE, "ClientKeyExchange"},
     {SSL3_MT_FINISHED, "Finished"},
+    {21, "CertificateUrl"},
     {SSL3_MT_CERTIFICATE_STATUS, "CertificateStatus"},
+    {23, "SupplementalData"},
```

Comment:
Why would we add these two? OpenSSL does not support either message type. Both message types will only appear if they have been negotiated via extensions - which will never happen if OpenSSL is one of the peers.

Reply:
To keep this list in sync with the one declared in apps\s_cb.c : https://github.com/openssl/openssl/blob/1d0c08b4963f5f7e1d1855e360417a11973d8455/apps/s_cb.c#L538

Reply:
I'm not sure why we care if these are in sync?

Reply:
It is trace code, so the input data and extensions will also come from outside.

Reply:
I agree, we should be able to prevent every packet type, if not its contents, even if we don't handle it.  For debugging.

Reply:
Hmm. I suppose if you were talking to a misbehaving peer it could send one of these messages in error. Or if you were using SSL_trace on a communication that did not involve OpenSSL itself as one of the peers. Neither seems especially likely - but possible. If we're going to do that then I would suggest we define the proper constants for these in the header files.